src/mxcp/server/interfaces/server/mcp.py

```diff
@@ -562,13 +560,36 @@ def _register_signal_handlers(self) -> None:
 
     def _handle_reload_signal(self, signum: int, frame: Any) -> None:
         """Handle SIGHUP signal to reload the configuration."""
-        logger.info("Received SIGHUP signal, initiating configuration reload...")
+        logger.info("Received SIGHUP signal, scheduling configuration reload...")
+
+        loop = self._signal_loop
+        if isinstance(loop, asyncio.AbstractEventLoop) and loop.is_running():
+            loop.call_soon_threadsafe(
+                lambda: asyncio.create_task(self._handle_reload_signal_async())
+            )
+        else:
+            logger.warning("Event loop not ready for async reload; running blocking reload handler")
```

Comment:
Why that logic? How an event loop could be not ready?

Reply:
Good catch. It is a leftover from the step-by-step migration.